### PR TITLE
Document the transaction fields

### PR DESCRIPTION
**Pull request by turbolent**
It was not clear what the block reference field means. I copied variants of the comments from https://github.com/dapperlabs/flow-go/blob/25a69c092fca3efe3f2c140c5d0acbff4e391b3e/model/flow/transaction.go

### CODE REVIEW

File: transaction.go [hunk 1]
```diff
@@ -30,14 +30,35 @@ import (
...
+	// This reference can be set to an older block if the tx should expire faster.
+	ReferenceBlockID Identifier
+
+	// GasLimit is the max amount of computation which is allowed to be used during this transaction
+	GasLimit uint64
+
+	// ProposalKey is key of the account used to propose the transaction
+	ProposalKey ProposalKey
+
+	// Payer is the account that pays the feed for this transaction
```
Comment (joshuahannan) [inline transaction.go:53]:
```suggestion
	// Payer is the account that pays the fee for this transaction
```

File: transaction.go [hunk 2]
```diff
@@ -30,14 +30,35 @@ import (
...
+	Payer Address
+
+	Authorizers []Address
+
+	// PayloadSignatures is the list of account signatures excluding signature of the payer account
+	PayloadSignatures []TransactionSignature
+
+	// EnvelopeSignatures is the payer signature over the envelope (payload + payload signatures)
```
Comment (joshuahannan) [inline transaction.go:61]: Could you clarify this a bit? Should it be `payer + payload` signatures?

**Comment by cybercent** [comment]
One field that needs a bit of clarification is `Authorizers`

File: transaction.go (hunk 2 above)
Comment (turbolent) [reply transaction.go:61]: Sorry, I just copied this over from the flow-go codebase

Comment (joshuahannan) [reply transaction.go:61]: I think I understand what it means now. I thought it was saying (payload signature + payload signature) and I thought it was a typo. But is actually saying (the entire payload + the signature for the payload).

**Comment by turbolent** [comment]
@psiemens would you be able to add a comment for `Authorizers`? the one in flow-go isn't really worth copying over

**Comment by psiemens** [comment]
@turbolent I updated the comments (8c4d958) to add a bit more clarity -- I found that the ones from `flow-go` weren't quite descriptive enough for people unfamiliar with Flow internals. I also added links to our new docs site for those who want to read more.

**Comment by turbolent** [comment]
Nice, so much better, looks good! Needs an approval from an owner and then I can merge.

**Review by psiemens — APPROVED** [review]
🚢

**Pull request merged**